tpot/tpot.py

```diff
@@ -288,6 +289,182 @@ def score(self, training_features, training_classes, testing_features, testing_c
 
         return self._evaluate_individual(self.optimized_pipeline_, training_testing_data)[0]
 
+    def export(self, output_file_name):
+        """Exports the current optimized pipeline as Python code.
+
+        Parameters
+        ----------
+        output_file_name: string
+            String containing the path and file name of the desired output file
+
+        Returns
+        -------
+        None
+        """
+        if self.optimized_pipeline_ == None:
+            raise Exception('A pipeline has not yet been optimized. Please call fit() first.')
+        
+        exported_pipeline = self.optimized_pipeline_
+        
+        # Replace all of the mathematical operators with their results
+        while True:
```

Comment:
I *think* you're right that it's unnecessary to repeatedly start over when looping through the TPOT stack, but it is necessary to loop through the function tree backwards (which will answer your question below as well).

The TPOT stack is stored as a list of functions and primitives, so for example,

`random_forest(decision_tree(dataset, 1+1, 2+2), 3+3, 4+(4+4)` would be stored as:

`[random_forest, decision_tree, dataset, +, 1, 1, +, 2, 2, +, 3, 3, +, +, 4, 4, +, 4, 4]`

Thus you can't run through the stack forwards because the parameters to the first functions have not been completely evaluated yet.

Therefore to deconstruct the TPOT stack, you need to proceed backwards through the stack and evaluate the functions as you run into them.

So proceeding backwards, you would first run into the `+`. You know that `+` accepts two parameters, so you take the two parameters that follow -- `4` and `4`. Evaluate `4+4` and replace those 3 entries by the result, `8`. Now you have:

`[random_forest, decision_tree, dataset, +, 1, 1, +, 2, 2, +, 3, 3, +, +, 4, 4, 8]`

You again run into a `+`, the following two parameters are `4` and `4`, repeat above...

`[random_forest, decision_tree, dataset, +, 1, 1, +, 2, 2, +, 3, 3, +, 8, 8]`

... and I think you get it from there. That leaves the following stack once all the mathematical operators are replaced:

`[random_forest, decision_tree, dataset, 2, 4, 6, 16]`

Next you run into `decision_tree`. You know that it takes 3 parameters, so you evaluate `decision_tree(dataset, 2, 4)` and replace its entry with the result (say, `result1`):

`[random_forest, result1, 6, 16]`

Then you finally have `random_forest`, which similarly takes 3 parameters, and you can evaluate the last call of the stack: `random_forest(result1, 6, 16)`

Make sense?